src/state.rs

```diff
@@ -172,13 +170,14 @@ impl<R: bls::rand::RngCore + Clone> DkgState<R> {
     }
 
     /// Handles the Parts to create the Acks
-    fn parts_into_acks(&mut self, parts: BTreeSet<IdPart>) -> Result<DkgVote> {
+    fn parts_into_acks<R: bls::rand::RngCore>(
+        &mut self,
+        parts: BTreeSet<IdPart>,
+        rng: &mut R,
```

Comment:
Again, should be able to change this to `mut rng: R` here. It makes for a slightly cleaner user API since the user won't have to take a mutable borrow for many cases.